### PR TITLE
Updating the pom.xml to use the nexus plugin and doing some refactoring to fix the failing tests

### PR DESCRIPTION
**Pull request by marcos-albornoz**
Updating the pom.xml to use the nexus plugin instead of the built in maven "deploy" plugin. Also added the correct support to use the gpg plugin to sign the artifacts and the use of the Envrironment Variable "ARTIFACT_RELEASE_VERSION" (which will be provided by Jenkins) to define the version of the artifacts.
Additionally it was refactored the method **_retrieveCertificate_** of _**CloudConnector**_ class to fix the test phase which ramdomly was failing and it was refactored the way that the exceptions on that class was created.

### CODE REVIEW

**Review by ricrodriguezg — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -270,19 +259,31 @@
...
 			</plugin>
+			<plugin>
+      			<groupId>org.sonatype.plugins</groupId>
+      			<artifactId>nexus-staging-maven-plugin</artifactId>
+      			<version>1.6.7</version>
+      			<extensions>true</extensions>
+      			<configuration>
+        			<serverId>ossrh</serverId>
+        			<nexusUrl>https://s01.oss.sonatype.org/</nexusUrl>
+        			<!-- skipStagingRepositoryClose>true</skipStagingRepositoryClose -->
```
Comment (angelmoo) [inline pom.xml:283]: are we going to use this on the future or we can safely remove it?

**Review by tr1ck3r — APPROVED** [review]
@marcos-albornoz can we use this opportunity to make sure we've been consistent with tab or space indenting throughout the `pom.xml` file?  Looks like we currently have a mix.

File: pom.xml (hunk 1 above)
Comment (marcos-albornoz) [reply pom.xml:283]: It's now disabled :-)

**Comment by marcos-albornoz** [comment]
> @marcos-albornoz can we use this opportunity to make sure we've been consistent with tab or space indenting throughout the `pom.xml` file? Looks like we currently have a mix.

I prefer to wait at the next release because if I indent correctly the whole pom.xml now, it will be a little confuse to difference the lines with changes which I want that you focus to review.

**Review by angelmoo — APPROVED** [review]
Changes looks good

**Review by luispresuelVenafi — APPROVED** [review]

**Pull request merged**